File: Blind_75_List/Queue/649. Dota2 Senate.py
# Problem Link: https://leetcode.com/problems/dota2-senate/
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        n = len(senate)
        l = collections.deque()
        r = collections.deque()
        for i,c in enumerate(senate):
            if c == 'R':
                r.append(i);
            else:
                l.append(i)
        while l and r:
            if l[0] < r[0]:
                r.popleft()
                l.append(l.popleft() + n)
            else:
                l.popleft()
                r.append(r.popleft() + n)
        
        if l:
            return "Dire"
        else:
            return "Radiant"
        
        

File: Blind_75_List/Queue/649. Dota2 Senate.cpp
//Problem Link: https://leetcode.com/problems/dota2-senate/
class Solution {
public:
    string predictPartyVictory(string senate) {
        queue<int> r,l;
        int n = senate.size();
        for(int i = 0; i<senate.size(); i++){
            if (senate[i] == 'R')
                r.push(i);
            else
                l.push(i);
        }
        while(!r.empty() && !l.empty()){
            if (r.front() < l.front()){
                l.pop();
                r.push(r.front() + n);
                r.pop();
            }
            else{
                r.pop();
                l.push(l.front() + n);
                l.pop();
                
            }

        }
        
        if (l.empty()){
            return "Radiant";
        }
        else
            return "Dire";
        
    }
};